### PR TITLE
[TOPI] LRN & L2norm Operator

### PR DESCRIPTION
**Pull request by PariksheetPinjari909**
LRN operator for caffe alexnet model to use with onnx frontend

@tqchen @sxjscience  your review comments from https://github.com/dmlc/tvm/pull/1000 is addressed here. https://github.com/dmlc/tvm/pull/1000 got accidentally closed.

### CODE REVIEW

**Comment by tqchen** [comment]
cc @sxjscience can you do another round of review?

**Comment by sxjscience** [comment]
Need to work on the project in Amazon. I'll have a deep dive into how to implement it better on Thursday. For now, I think we need to support L2Normalization with arbitrary axis.

**Comment by PariksheetPinjari909** [comment]
@sxjscience , shall I follow https://www.tensorflow.org/api_docs/python/tf/nn/l2_normalize   for l2norm implementation?

**Comment by PariksheetPinjari909** [comment]
@sxjscience , I have added arbitrary axis support for L2norm. Could you please help to review further.

**Comment by sxjscience** [comment]
I think it's good now. We can refactor the implementation later.

File: topi/python/topi/nn/local_response_norm.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    assert (size % 2) == 1, "size should be odd number"
+
+    ##Add padding on left & right of size radius first
+    pad_before = [0, (size//2), 0, 0]
+    pad_after = [0, (size//2), 0, 0]
+    pad_data = pad(data, pad_before, pad_after, name="pad_data")
+
+    rxk = tvm.reduce_axis((0, size), name='rxk')
+    sqr_sum = tvm.compute((b, c, h, w), lambda i, l, j, k: tvm.sum(
+        tvm.power(pad_data[i, l + rxk, j, k], 2.0),
```
Comment (tqchen) [inline topi/python/topi/nn/local_response_norm.py]: do not use power to calculate the square, instead use x* x

File: topi/tests/python/test_topi_lrn.py [hunk 2]
```diff
@@ -29,45 +32,53 @@ def lrn_nchw_python(a_np, size, bias, alpha, beta):
...
+                                           (a_np[i, j + rxj, k, l] * \
+                                            a_np[i, j + rxj, k, l])
+        elif (axis == 3):
+            #NHWC layout
+            for rxl in range(-radius, radius+1):
+                if ((l + rxl) < axis3) and ((l + rxl) >= 0):
+                    sqr_sum[i, j, k, l] = sqr_sum[i, j, k, l] + \
+                                           (a_np[i, j, k, l + rxl] * \
+                                            a_np[i, j, k, l + rxl])
+    for i in range(axis0):
```
Comment (tqchen) [inline topi/tests/python/test_topi_lrn.py]: use broadcasting semantics, instead of the loop

**Comment by tqchen** [comment]
Thanks for improving the code during the review process, this is now merged!

**Pull request merged**